dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/handler/ExternalSourceEntryUriListHandler.java

```diff
@@ -0,0 +1,139 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest.repository.handler;
+
+import java.sql.SQLException;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.servlet.http.HttpServletRequest;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.Logger;
+import org.dspace.authorize.AuthorizeException;
+import org.dspace.authorize.service.AuthorizeService;
+import org.dspace.content.Collection;
+import org.dspace.content.Item;
+import org.dspace.content.service.CollectionService;
+import org.dspace.core.Context;
+import org.dspace.external.model.ExternalDataObject;
+import org.dspace.external.service.ExternalDataService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.rest.webmvc.ResourceNotFoundException;
+import org.springframework.stereotype.Component;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+/**
+ * This class will handle ExternalSourceEntryUriList and it'll create Item objects based on them
+ */
+@Component
+public class ExternalSourceEntryUriListHandler implements UriListHandler<Item> {
+
+    private List<RequestMethod> allowedRequestMethods = new LinkedList<>(Arrays.asList(RequestMethod.POST));
+
+    @Autowired
+    private AuthorizeService authorizeService;
+
+    @Autowired
+    private ExternalDataService externalDataService;
+
+    @Autowired
+    private CollectionService collectionService;
+
+    private static final Logger log = org.apache.logging.log4j.LogManager
+        .getLogger(ExternalSourceEntryUriListHandler.class);
+
+
+    @Override
+    public boolean supports(List<String> uriList, String method) {
+        if (!allowedRequestMethods.contains(RequestMethod.valueOf(method))) {
+            return false;
+        }
+        for (String string : uriList) {
+            if (!(StringUtils.contains(string, "api/integration/externalsources") &&
```

Comment:
Rather than hardcoding this string path, shouldn't we be using `ExternalSourceRest.CATEGORY` and  `ExternalSourceRest.PLURAL_NAME`?